### PR TITLE
Update to Perks to add Type.

### PR DESCRIPTION
**Pull request by JFranklin3000**
Added Perk Type to the Perk Item Sheet.

### CODE REVIEW

**Review by phildominguez — CHANGES_REQUESTED** [review]
image
Game System and Type should show up in some capacity when the item is expanded on the actor sheet. Probably as chips?
image
Similar for the chat card, though now I'm noticing Source is broken there.

**Comment by phildominguez** [comment]
Also just wanted to confirm that System should be checkboxes and not a selector? Are there Perks shared between the systems?

**Comment by JFranklin3000** [comment]
> Also just wanted to confirm that System should be checkboxes and not a selector? Are there Perks shared between the systems?
image

image

image
 
First General Perk in the 3 existing games

**Comment by phildominguez** [comment]
> 

Gotcha, thanks for the example!

File: module/helpers/config.mjs [hunk 1]
```diff
@@ -290,6 +290,15 @@ E20.lightRanges = {
...
 
+// Perk types
+E20.perkType = {
+  general: "E20.PerkGeneral",
+  influence: "E20.PerkInfluence",
+  origin: "E20.PerkOrigin",
+  role: "E20.PerkRole"
+};
+
+
```
Comment (phildominguez) [inline module/helpers/config.mjs:301]: Remove extra newline

File: template.json [hunk 2]
```diff
@@ -483,7 +483,8 @@
       "templates": [
         "itemDescription"
       ],
-      "prerequisite": null
+      "prerequisite": null,
+      "type": "General"
```
Comment (phildominguez) [inline template.json:487]: This should be `"general"` to match the key in config.mjs

File: templates/item/item-perk-sheet.hbs [hunk 3]
```diff
@@ -12,12 +12,18 @@
...
+    {{#*inline "item-field-inputs"}}
+    <select class="item-field-inputs" name="system.perkType">
+      {{#select system.perkType}}
+      {{#each config.perkType as |name type|}}
+      <option value="{{type}}">{{localize name}}</option>
+      {{/each}}
+      {{/select}}
+    </select>
+    {{/inline}}
+    {{/"systems/essence20/templates/item/parts/item-sheet-field.hbs"}}
```
Comment (phildominguez) [inline templates/item/item-perk-sheet.hbs:26]: Indentation is off for lines within the div

**Review by phildominguez — CHANGES_REQUESTED** [review]
Couple small things 👍

File: templates/item/item-perk-sheet.hbs (hunk 3 above)
Comment (JFranklin3000) [reply templates/item/item-perk-sheet.hbs]: fixed

File: template.json (hunk 2 above)
Comment (JFranklin3000) [reply template.json]: changed

File: module/helpers/config.mjs (hunk 1 above)
Comment (JFranklin3000) [reply module/helpers/config.mjs]: removed

File: templates/item/item-perk-sheet.hbs [hunk 4]
```diff
@@ -24,6 +25,15 @@
...
       {{/"systems/essence20/templates/item/parts/item-sheet-field.hbs"}}
+      
+      {{!-- Prerequisite --}}
+      {{#> "systems/essence20/templates/item/parts/item-sheet-field.hbs" label='E20.PerkPrerequisite'}}
+      {{#*inline "item-field-inputs"}}
+      <input type="text" name="system.prerequisite" value="{{system.prerequisite}}" />
+      {{/inline}}
+      {{/"systems/essence20/templates/item/parts/item-sheet-field.hbs"}}
+
+
```
Comment (phildominguez) [inline templates/item/item-perk-sheet.hbs:36]:
```suggestion
```

File: templates/item/item-perk-sheet.hbs [hunk 5]
```diff
@@ -12,6 +12,7 @@
   {{!-- Item Sheet Body --}}
   <section class="sheet-body item-sheet">
     <div class="tab main flexcol" data-group="primary" data-tab="main">
+
```
Comment (phildominguez) [inline templates/item/item-perk-sheet.hbs:15]:
```suggestion
```
Nbd to me if there's a newline here or not, but there's no corresponding newline where the div closes.

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: templates/item/item-perk-sheet.hbs (hunk 5 above)
Comment (JFranklin3000) [reply templates/item/item-perk-sheet.hbs:15]: Changes made

File: templates/item/item-perk-sheet.hbs (hunk 4 above)
Comment (JFranklin3000) [reply templates/item/item-perk-sheet.hbs]: changes made

**Review by phildominguez — APPROVED** [review]

**Pull request merged**